test/Microsoft.IdentityModel.Tokens.Tests/Base64UrlEncodingTests.cs

```diff
@@ -309,23 +309,23 @@ public Base64UrlEncoderTheoryData(string testId) : base(testId) { }
         }
 
         [Fact]
-        public void ValidateAndGetOutputSizeTests()
+        public void ValidateTests()
         {
             string input = string.Empty;
-            Assert.Throws<ArgumentNullException>(() => Base64UrlEncoding.ValidateAndGetOutputSize(input.AsSpan(), 0, 0));
-            Assert.Throws<ArgumentException>(() => Base64UrlEncoding.ValidateAndGetOutputSize("abc".AsSpan(), -1, 3));
-            Assert.Throws<ArgumentException>(() => Base64UrlEncoding.ValidateAndGetOutputSize("abc".AsSpan(), 0, -1));
-            Assert.Throws<ArgumentException>(() => Base64UrlEncoding.ValidateAndGetOutputSize("abc".AsSpan(), 0, 4));
-            Assert.Throws<FormatException>(() => Base64UrlEncoding.ValidateAndGetOutputSize("abcde".AsSpan(), 0, 5));
+            Assert.Throws<ArgumentNullException>(() => Base64UrlEncoding.Validate(input.AsSpan(), 0, 0));
+            Assert.Throws<ArgumentException>(() => Base64UrlEncoding.Validate("abc".AsSpan(), -1, 3));
+            Assert.Throws<ArgumentException>(() => Base64UrlEncoding.Validate("abc".AsSpan(), 0, -1));
+            Assert.Throws<ArgumentException>(() => Base64UrlEncoding.Validate("abc".AsSpan(), 0, 4));
+            Assert.Throws<FormatException>(() => Base64UrlEncoding.Validate("abcde".AsSpan(), 0, 5));
 
-            int actualOutputSize = Base64UrlEncoding.ValidateAndGetOutputSize("abc".AsSpan(), 0, 0);
+            int actualOutputSize = Base64UrlEncoding.Validate("abc".AsSpan(), 0, 0).Length;
             Assert.Equal(0, actualOutputSize);
 
-            actualOutputSize = Base64UrlEncoding.ValidateAndGetOutputSize("abcd".AsSpan(), 0, 4);
-            Assert.Equal(3, actualOutputSize);
```

Comment:
Hmm. Is this behavior wrong? Why would it return 3?

Reply:
The previous implementation had a different way of calculating the output size.

Reply:
What is the difference? Does the new implementation allocate more bytes then? Shouldn't the encoding algorithm be the same?

Reply:
The `Base64Url` API calculates the max possible size whereas the custom implementation here appears to be able to find the exact size of the decoded value.